src/domain/session/rightpanel/MemberListViewModel.js

```diff
@@ -0,0 +1,35 @@
+import {ViewModel} from "../../ViewModel.js";
+import {MemberTileViewModel} from "./MemberTileViewModel.js";
+import {createMemberComparator} from "./comparator.js";
+import {Disambiguator} from "./disambiguator.js";
+
+export class MemberListViewModel extends ViewModel {
+    constructor(options) {
+        super(options);
+        this.memberTileViewModels = this._mapTileViewModels(this._filterJoinedMembers(options.members))
+                                        .sortValues(createMemberComparator(options.powerLevels));
+        this.nameDisambiguator = new Disambiguator();
+        this.mediaRepository = options.mediaRepository;
+    }
+
+    get type() { return "member-list"; }
+
+    _filterJoinedMembers(members) {
+        return members.filterValues(member => member.membership === "join");
+    }
+
+    _mapTileViewModels(members) {
+        const mapper = (member, emitChange) => {
+            const mediaRepository = this.mediaRepository;
+            const vm = new MemberTileViewModel(this.childOptions({member, emitChange, mediaRepository}));
+            this.nameDisambiguator.disambiguate(vm);
+            return vm;
+        }
+        const updater = (vm, params, newMember) => {
+            vm.updateFrom(newMember);
+            this.nameDisambiguator.disambiguate(vm);
```

Comment:
Noticed something here. `MappedMap` won't run the mapper in `onRemove`. This will cause any user who leaves and rejoins a room to appear disambiguated.

Reply:
Ah yes, can you check for this in the mapper when it is run in onAdd?

Reply:
Fixed for now but there are related issues. For instance if user-A and user-B are disambiguated and user-B leaves the room, we'd ideally want user-A to be disambiguated (which we can't really do atm because the disambiguator does not get informed when user-B leaves).

Reply:
good point, I think this is fine though for now.